cli/azd/docs/experimental-features.md

```diff
@@ -0,0 +1,54 @@
+# Experimental Features
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+- Enable _all_ experimental features: `azd config set experimental on`
+- Enable _specific_ experimental feature: `azd config set experimental.name.feature`
```

Comment:
`azd config set` takes two arguments, a path and a value.  When paths have dots in them, it forms a JSON object behind the scenes for each node.

In the example for enabling a specific feature, you don't have the value, but based on the above I think you might have intended:

`azd config set experimental.name.feature on`

The use of dots in your example makes me think that the value of experimental should be an object, so the first example where we set the all experimental features to on should perhaps be: `azd config set experimental.all on`?

Another question: is "name.feature" an "atomic unit" here?  Put another way, do we always expect when you enable a specific experiment, you have a path with two dots in it?  Imagine that we were adding the `pulumi` experiment,   would we do `azd config set experimental.pulumi on` or `azd config set experimental.name.pulumi on`  something else?

Reply:
> Another question: is "name.feature" an "atomic unit" here?

No, the key is just a unique string to identify the `alpha-feature`.  I think the use of a `dot` is recognized by the `config` command as the structure-level for the persisted json file.  But that's an implementation detail which should not matter to users.